### PR TITLE
Adding API comparison and Warnings

### PR DESCRIPTION
**Pull request by KanishkT123**
Issue Number: (Link to Github Issue or Azure Dev Ops Task/Story)

## Summary
Adds the Open API generation tool to compare and warn reviewers about API changes.

## Changes

- New Job added to make API comparison from current branch to dev
- Warning added when API comparison is non-0
- Changes to PR template to clarify when API changes are needed.

## Checklist

- [x] Local Tests Passing?
- [x] CICD and Pipeline Tests Passing?
- [x] Added any new Tests?
- [x] Documentation Updates Made?
- [x] This is not a breaking change. If it is, please describe it below.

## Is this a breaking change?
No.

## Anything else?
Other comments, collaborators, etc.
CC: @pritipath @juzuluag

### CODE REVIEW

File: .github/workflows/build-webapi.yaml [hunk 1]
```diff
@@ -37,11 +37,11 @@ jobs:
       - name: Build
         run: dotnet build --configuration Release --no-restore
         working-directory: ${{ env.DOTNET_SRC_DIR }}
-      - name: Test + Code Coverage
-        run: dotnet test --no-restore --verbosity normal /p:CollectCoverage=true /p:CoverletOutputFormat=opencover
-        working-directory: ${{ env.DOTNET_SRC_DIR }}
-      - name: Codecov
-        uses: codecov/codecov-action@v1
+      # - name: Test + Code Coverage
```
Comment (rakker91) [inline .github/workflows/build-webapi.yaml]: did you mean to kill code coverage?

**Review by rakker91 — APPROVED** [review]

File: .github/workflows/build-webapi.yaml [hunk 2]
```diff
@@ -74,3 +74,38 @@ jobs:
...
           wget -t 5 --waitretry=5 ${DATA_ENDPOINT}
 
+  api-comparison:
+      needs: container-dotnet-build
+      runs-on: ubuntu-latest
+      defaults:
+        run:
+          working-directory: ./src/CarbonAware.WebApi/src
+      container:
+        image: mcr.microsoft.com/dotnet/sdk
```
Comment (juzuluag) [inline .github/workflows/build-webapi.yaml:84]: nit: Does it make sense to pin the version here? `mcr.microsoft.com/dotnet/sdk:6.0`

**Review by juzuluag — APPROVED** [review]
Just a comment, other than that lgtm.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -13,8 +13,12 @@ One sentence summary of PR
 - [ ] CICD and Pipeline Tests Passing?
 - [ ] Added any new Tests?
 - [ ] Documentation Updates Made?
+- [ ] Are there any API Changes? If yes, pelase describe below.
```
Comment (bderusha) [inline .github/pull_request_template.md]:
```suggestion
- [ ] Are there any API Changes? If yes, please describe below.
```

Comment (bderusha) [reply .github/pull_request_template.md]: Consider linking the action that is related to this: https://github.com/microsoft/carbon-aware-sdk/actions/workflows/build-webapi.yaml

**Review by bderusha — APPROVED** [review]
Commented on some small bits, but otherwise looks good!

**Pull request merged**